openstates/va/csv_bills.py

```diff
@@ -0,0 +1,368 @@
+
+    def load_amendments(self):
+        resp = self.get(self._url_base + "Amendments.csv").text
+        reader = csv.reader(resp.splitlines(), delimiter=",")
+
+        # ['BILL_NUMBER', 'TXT_DOCID']
+        for row in reader:
+            self._amendments[row[0].strip()].append(
+                {"bill_number": row[0].strip(), "txt_docid": row[1].strip()}
+            )
+        self.warning("Total Amendments Loaded: " + str(len(self._amendments)))
+
+    def load_history(self):
+        resp = self.get(self._url_base + "HISTORY.CSV").text
+        reader = csv.reader(resp.splitlines(), delimiter=",")
+        # ['Bill_id', 'History_date', 'History_description', 'History_refid']
+        for row in reader:
+            self._history[row[0]].append(
+                {
+                    "bill_id": row[0],
+                    "history_date": row[1],
+                    "history_description": row[2],
+                    "history_refid": row[3],
+                }
+            )
+        self.warning("Total Actions Loaded: " + str(len(self._history)))
+
+    def load_votes(self):
+        resp = self.get(self._url_base + "VOTE.CSV").text.splitlines()
+        for line in resp:
+            line = line.split(",")
+            # First part of the line is always the history_refid number.
+            #   It has extra quotes around it
+            # Next number is the member_id number found in _members
+            # Y and X represent yes or no
+            history_refid = line[0].replace('"', "")
+            # Checks if votes are present
+            if len(line) > 1:
+                # Not every line has the same number of votes.
+                for v in range(1, len(line), 2):
+                    if (
+                        line[v] != '"H0000"'
+                        and len(self._members[line[v].replace('"', "")]) > 0
+                    ):
+                        member = self._members[line[v].replace('"', "")][0]["name"]
+                        vote_result = line[v + 1].replace('"', "")
+                        if vote_result == "Y":
+                            vote_result = "yes"
+                        elif vote_result == "N":
+                            vote_result = "no"
+                        elif vote_result == "X":
+                            vote_result = "not voting"
+                        elif vote_result == "A":
+                            vote_result = "abstain"
+                        self._votes[history_refid].append(
+                            {"member_id": member, "vote_result": vote_result}
+                        )
+        self.warning("Total Votes Loaded: " + str(len(self._votes)))
+
+    def load_bills(self):
+        resp = self.get(self._url_base + "BILLS.CSV").text
+        reader = csv.reader(resp.splitlines(), delimiter=",")
+        # ["Bill_id" (0), "Bill_description (1)", "Patron_id (2)", "Patron_name (3)", "Last_house_committee_id (4)",
```

Comment:
let's make sure to comment where these come from

for this many fields thinking it might make the code more clear overall if we use DictReader & pass in the field list

https://docs.python.org/3/library/csv.html#csv.DictReader

Reply:
Switched over to using csv.DictReader()